enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,491 @@
+   - Can we take advantage of or replicate the installer's [bootstrap
+     service
+     records](https://github.com/openshift/installer/blob/6d778f911e79afad8ba2ff4301eda5b5cf4d8e9e/docs/dev/bootstrap_services.md)
+     behavior?
+
+2. Is the workflow different for single-node? Do we need to start the
+   service and go through the orchestration steps for one node? If we
+   don't, how do we handle the validations?
+3. How do we implement a feature flag in the installer?
+
+### Test Plan
+
+We will implement CI jobs to verify deployment of clusters in all
+three topologies (single-node, compact, and HA) using VMs simulating
+bare metal hosts. We will configure the jobs so that they run for
+changes in the dev-scripts, installer, assisted-service, and
+assisted-agent repositories to ensure that regressions are not
+introduced.
+
+We will implement end-to-end CI jobs to exercise the
+parallel/serial/upgrade suites running periodically as release
+informers/blockers to ensure that clusters created by the agent-based
+installer meet the expectations of all OpenShift clusters.
+
+We will use unit and functional tests on the installer repository to
+test the image generator using different types of inputs.
+
+### Graduation Criteria
+
+We will use a fork or branch of the installer to prototype this work,
+and then merge it into the installer repository protected by a feature
+flag. The minimum viable version of the new feature will be exposed as
+tech preview (deployment failures may not receive immediate attention,
+but clusters successfully deployed using the tool will be fully
+supported).
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- The minimum viable version of the new installer may only work in
+  connected settings
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+- Must support deploying in fully disconnected settings
+- Automated end-to-end tests
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+Does not apply for the installer.
+
+### Version Skew Strategy
+
+Does not apply for the installer.
+
+### Operational Aspects of API Extensions
+
+No API changes.
+
+#### Failure Modes
+
+No API changes.
+
+#### Support Procedures
+
+No API changes.
+
+## Implementation History
+
+A prototype of the `create image` command is available in the
+`fleeting` repository:
+https://github.com/openshift-agent-team/fleeting
+
+## Drawbacks
+
+N/A
+
+## Alternatives
```

Comment:
Run the assisted service outside the installed cluster

Reply:
That was covered extensively in the discussion on https://github.com/openshift/enhancements/blob/master/enhancements/agent-installer/agent-based-installer.md